src/analysis/inner_product_space/adjoint.lean

```diff
@@ -0,0 +1,145 @@
+begin
+  refine le_antisymm _ _,
+  { refine continuous_linear_map.op_norm_le_bound _ (norm_nonneg _) (λ x, _),
+    rw [adjoint'_apply, linear_isometry_equiv.norm_map],
+    exact to_sesq_form_apply_norm_le },
+  { nth_rewrite_lhs 0 [←adjoint'_adjoint' A],
+    refine continuous_linear_map.op_norm_le_bound _ (norm_nonneg _) (λ x, _),
+    rw [adjoint'_apply, linear_isometry_equiv.norm_map],
+    exact to_sesq_form_apply_norm_le }
+end
+
+/-- The adjoint of a bounded operator from Hilbert space E to Hilbert space F. -/
+def adjoint : (E →L[𝕜] F) ≃ₗᵢ⋆[𝕜] (F →L[𝕜] E) :=
+linear_isometry_equiv.of_surjective
+{ norm_map' := adjoint'_norm,
+  ..adjoint' }
+(λ A, ⟨adjoint' A, adjoint'_adjoint' A⟩)
+
+local postfix `†`:1000 := adjoint
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_left (A : E →L[𝕜] F) (x : E) (y : F) : ⟪A† y, x⟫ = ⟪y, A x⟫ :=
+adjoint'_inner_left A x y
+
+/-- The fundamental property of the adjoint. -/
+lemma adjoint_inner_right (A : E →L[𝕜] F) (x : E) (y : F) : ⟪x, A† y⟫ = ⟪A x, y⟫ :=
+adjoint'_inner_right A x y
+
+/-- The adjoint is involutive -/
+@[simp] lemma adjoint_adjoint (A : E →L[𝕜] F) : A†† = A :=
+adjoint'_adjoint' A
+
+/-- The adjoint of the composition of two operators is the composition of the two adjoints
+in reverse order. -/
+@[simp] lemma adjoint_comp (A : F →L[𝕜] G) (B : E →L[𝕜] F) : (A ∘L B)† = B† ∘L A† :=
+begin
+  ext v,
+  refine ext_inner_left 𝕜 (λ w, _),
+  simp only [adjoint_inner_right, continuous_linear_map.coe_comp', function.comp_app],
+end
+
+/-- `E →L[𝕜] E` is a C⋆-algebra with the adjoint as the star operation. -/
+instance : has_star (E →L[𝕜] E) := ⟨adjoint⟩
+instance : has_involutive_star (E →L[𝕜] E) := ⟨adjoint_adjoint⟩
+instance : star_monoid (E →L[𝕜] E) := ⟨adjoint_comp⟩
+instance : star_ring (E →L[𝕜] E) := ⟨linear_isometry_equiv.map_add adjoint⟩
+instance : star_module 𝕜 (E →L[𝕜] E) := ⟨linear_isometry_equiv.map_smulₛₗ adjoint⟩
+
+lemma star_eq_adjoint (A : E →L[𝕜] E) : star A = A† := rfl
+
+instance : cstar_ring (E →L[𝕜] E) :=
+⟨begin
+  intros A,
+  rw [star_eq_adjoint],
+  refine le_antisymm _ _,
+  { calc ∥A† * A∥ ≤ ∥A†∥ * ∥A∥      : op_norm_comp_le _ _
+              ... = ∥A∥ * ∥A∥       : by rw [linear_isometry_equiv.norm_map] },
+  { rw [←sq, ←real.sqrt_le_sqrt_iff (norm_nonneg _), real.sqrt_sq (norm_nonneg _)],
+    refine op_norm_le_bound _ (real.sqrt_nonneg _) (λ x, _),
+    have := calc
+      re ⟪A† (A x), x⟫ ≤ abs ⟪A† (A x), x⟫    : re_le_abs _
+                  ...  ≤ ∥A† (A x)∥ * ∥x∥     : abs_inner_le_norm _ _
+                  ...  ≤ ∥A† * A∥ * ∥x∥ * ∥x∥
+                    : begin
+                        refine mul_le_mul_of_nonneg_right _ (norm_nonneg _),
+                        rw [←comp_apply],
+                        exact le_op_norm _ _,
+                      end,
+    calc ∥A x∥ = real.sqrt (re ⟪A x, A x⟫)          : norm_eq_sqrt_inner _
+          ...  = real.sqrt (re ⟪A† (A x), x⟫)       : by rw [←adjoint_inner_left]
```

Comment:
Perhaps the fact `⟪(A† * A) x, x⟫ = ∥A x∥ ^ 2` is worth extracting as a lemma, I feel like I've seen this trick used many times.

Reply:
I just added a few versions of this.